src/tbp/monty/frameworks/utils/evidence_matching.py

```diff
@@ -429,8 +429,13 @@ def select_hypotheses(
         Args:
             slope_threshold: Minimum slope value to keep a removable (sufficiently old)
                 hypothesis.
+            min_maintained_hyps: Minimum number of hypotheses to maintain.
             channel: Name of the input channel.
 
+        Note that the parameter `min_maintained_hyps` overrides the `slope_threshold`
+        and the removable mask. These hypotheses will be maintained regardless of their
```

Comment:
```suggestion
        and the removable mask. These hypotheses will be maintained in spite of their
```

I think in spite of might be a bit clearer than "regardless", since the age and slope can in fact influence which ones are maintained to satisfy the `min_maintained_hyps`.

Reply:
Wording updated in [a6700264e441](https://github.com/thousandbrainsproject/feat.dynamic_resizing/commit/a6700264e4415dbacabfdc52b953502666e5479d)